packages/runed/src/lib/utilities/FiniteStateMachine/FiniteStateMachine.svelte.ts

```diff
@@ -0,0 +1,115 @@
+export class FiniteStateMachine<StatesT extends string, EventsT extends string> {
+	#current = $state() as StatesT;
```

Comment:
This actually doesn't work, because then current ends up with the type of `StatesT | undefined` which we don't want. @Hugos68 suggested the type assertion as a workaround and it's better than the alternative IMO!

Reply:
Sorry, for more background: I originally implemented this as a function that returned an object, but @Hugos68 recommended that I turn it into a class. But then, I have to declare the `current` member outside the initializer, and I don't have the `initial` value where I'm wrapping it in the `$state()` rune.

tl;dr this is kind of the cost of doing it as a class, seems like a reasonable tradeoff?